src/main/java/org/jsoup/helper/W3CDom.java

```diff
@@ -366,9 +406,9 @@ public void head(org.jsoup.nodes.Node source, int depth) {
                 tagname to something safe, because that isn't going to be meaningful downstream. This seems(?) to be
                 how browsers handle the situation, also. https://github.com/jhy/jsoup/issues/1093 */
                 try {
-                    Element el = namespace == null && tagName.contains(":") ?
-                        doc.createElementNS("", tagName) : // doesn't have a real namespace defined
-                        doc.createElementNS(namespace, tagName);
+                    // use an empty namespace if none is present but the tag name has a prefix
```

Comment:
I didn't change any functionality here. The modification makes it clear that the only thing happening is that you're "imputing" a namespace of the empty string under certain conditions — you're not calling a separate method. The existing code duplicated the call to `doc.createElementNS()`, which primarily was confusing and obscured the purpose of the logic.